docs/api-docs/storefront/widgets/widget-versioning-tutorial.md

```diff
@@ -0,0 +1,255 @@
+Host: api.bigcommerce.com
+X-Auth-Token: {{ACCESS_TOKEN}}
+Content-Type: application/json
+Accept: application/json
+
+{
+  "name": "Red Text",
+  "schema": [
+    {
+      "type": "tab",
+      "label": "Content",
+      "sections": [
+        {
+          "settings": [
+            {
+              "type": "text",
+              "label": "Text Content",
+              "id": "textContent",
+              "typeMeta": {
+                "maxLength": 1000
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  "template": "<h1 style='color:red;'>{{textContent}}</h1>",
+  "create_new_version": true
+}
+```
+
+[![Open in Request Runner](https://storage.googleapis.com/bigcommerce-production-dev-center/images/Open-Request-Runner.svg)](https://developer.bigcommerce.com/api-reference/store-management/widgets/widget-template/updatewidgettemplate#requestrunner)
+
+Let's look at our widget again. Send a `GET` request to [`/v3/content/widgets/{uuid}`](https://developer.bigcommerce.com/api-reference/store-management/widgets/widget/getwidget) to retrieve your widget's data.
+You will notice that `current_version_uuid` has changed and is now different from the widget's `version_uuid`. Although we have updated the widget template, the widget created using that template did not change.
+
+### Create a new widget
+
+All widgets created after the update will be based on the most current version. Let's create another widget to test this functionality. 
+
+Send a `POST` request to [`/v3/content/widgets`](https://developer.bigcommerce.com/api-reference/storefront/widgets-api/widget/createwidget) replacing the `widget_template_uuid` placeholder with your widget template `uuid`.
+
+```http
+POST /stores/{{STORE_HASH}}/v3/content/widgets
+Host: api.bigcommerce.com
+X-Auth-Token: {{ACCESS_TOKEN}}
+Content-Type: application/json
+Accept: application/json
+
+{
+  "name": "Version 2 Widget",
+  "widget_configuration": {
+    "textContent": "Hello, World!"
+  },
+  "widget_template_uuid": "{your-widget-template-uuid}"
+}
+```
+
+[![Open in Request Runner](https://storage.googleapis.com/bigcommerce-production-dev-center/images/Open-Request-Runner.svg)](https://developer.bigcommerce.com/api-reference/store-management/widgets/widget/createwidget#requestrunner)
+
+To display the new widget next to the old one, place it in the same region. In our case, it is `home_below_featured_products`. 
+
+You should now see two different widgets displayed below **Featured Products**.
+
+![Add a new widget](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/widget-versioning-02.png "Add a new widget")
+
+## Upgrade the widget
+
+To upgrade the widget derived from the original widget template, send `"upgrade":true` in a `PUT` request to [`/v3/content/widgets/{uuid}`](https://developer.bigcommerce.com/api-reference/store-management/widgets/widget/updatewidget). This will push the widget up to the latest widget template version.
```

Comment:
We might want to make a note here that the newer version of the WidgetTemplate might/can have a different schema settings and we should provide the necessary configuration values along with `upgrade` flag to make sure the Widget doesn't break once upgraded. 
Like say we have a new setting to increase the font or say you have a new title heading which might need a new configuration value when creating the Widget.
We can think of adding this comment as a future addition if the current document is not sufficient.